python/cugraph/cugraph/dask/centrality/betweenness_centrality.py

```diff
@@ -101,6 +124,7 @@ def betweenness_centrality(
         int, list, cudf.Series, cudf.DataFrame, dask_cudf.Series, dask_cudf.DataFrame
     ] = None,
     normalized: bool = True,
+    weight: cudf.DataFrame = None,
```

Comment:
But other algorithms like `pagerank` have unused arguments([dangling](https://github.com/rapidsai/cugraph/blob/branch-23.08/python/cugraph/cugraph/link_analysis/pagerank.py#L167)) that are just there for NX compatibility. Wasn't it the intention when adding the argument `weights`

Reply:
Hmmm, I'm not confident we'd be adding much actual Nx compatibility with keeping `weight` as described here.  The Nx weight parameter for BC is a string to specify the weight attribute on a Nx Graph, but this is a Dataframe with the weight values.  I'm also curious about when we support it why we wouldn't just use the weights on a weighted graph instead of passing in a separate dataframe (which is much closer to Nx's API)?

Reply:
@jnke2016 , @ChuckHastings , and I discussed this offline and concluded we'd leave this in place since it now makes the MG API match the SG API.  One other aspect is the fact that the `weight` param is a dataframe though, and we decided that when/if we support weights in the future we'd likely change it to something that can take advantage of weights already on the graph, or something that takes advantage of the new edge ID support, and/or possibly something related to PropertyGraph support.